core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1187,22 +1190,43 @@ private ExprValue exprDayOfYear(ExprValue date) {
     return new ExprIntegerValue(date.dateValue().getDayOfYear());
   }
 
+  /**
+   * Implements extract function. Returns a LONG formatted according to the 'part' argument.
+   *
+   * @param part Literal that determines the format of the outputted LONG.
+   * @param datetime The date/datetime to be formatted.
+   * @return A LONG
+   */
+
   private ExprValue exprExtract(ExprValue part, ExprValue datetime) {
     String partName = part.stringValue().toUpperCase();
 
-    if(partName == "QUARTER"){
-      return exprQuarter(datetime);
-    }
+    long formattedVal = Long.parseLong(DateTimeFormatterUtil.getFormattedDate(
+        datetime,
+        new ExprStringValue(extract_formats.get(partName))).stringValue());
 
-    if (extract_formats.containsKey(partName)) {
-      long formattedVal = Long.parseLong(DateTimeFormatterUtil.getFormattedDate(
-          datetime,
-          new ExprStringValue(extract_formats.get(partName))).stringValue());
+    return new ExprLongValue(formattedVal);
+  }
 
-      return new ExprLongValue(formattedVal);
-    }
+  /**
+   * Implements extract function. Returns a LONG formatted according to the 'part' argument.
+   *
+   * @param part Literal that determines the format of the outputted LONG.
+   * @param time The time to be formatted.
+   * @return A LONG
+   */
+  private ExprValue exprExtractForTime(
+      FunctionProperties functionProperties,
+      ExprValue part,
+      ExprValue time) {
+    String partName = part.stringValue().toUpperCase();
+    LocalDateTime now = extractDateTime(time, functionProperties);
+
+    long formattedVal = Long.parseLong(DateTimeFormatterUtil.getFormattedDate(
+        new ExprDatetimeValue(now),
+        new ExprStringValue(extract_formats.get(partName))).stringValue());
```

Comment:
You can extract common code piece to another function. The diff is actually in `new ExprDatetimeValue(extractDateTime(time, functionProperties))`.

Reply:
Refactored [here](https://github.com/Bit-Quill/opensearch-project-sql/commit/4dccd2358d4f372469933ef1d5deea9125b4e890#diff-d90373fdb80f8b84b634db1be73e3689479691b99650aef7a2050057a29b956f)